### ISSUE TITLE
Not suspending tabs or saving settings

### ISSUE THREAD

**Issue by norachuga** [deanoemcke/thegreatsuspender#167]
Version is 6.15. I was seeing the same behavior on 6.13 and updated to latest. 

Tabs do not suspend at all. Neither through the icon menu, keyboard shortcut, or time.
This is the error that appears when I open Inspect Popup.
Uncaught TypeError: Cannot call method 'requestTabInfo' of undefined - popup.js:128

Settings will not save if you adjust the automatic suspension time.
Uncaught TypeError: Cannot call method 'resetAllTabTimers' of undefined - options.js:136
Refused to execute JavaScript URL because it violates the following Content Security Policy directive: "script-src 'self' https://ssl.google-analytics.com". - options.html:1

This is in Chrome Version 33.0.1750.117 m. I just now realized how old that is, but I have been unable to update it due to corporate policy. I'm still submitting this issue because I see reported issues for offbeat browsers like Opera & Aviator, and it might help diagnose the issue to know it is failing in an older Chrome as well.

**Comment by norachuga** [comment]
Alright, put my mind to it and got around corporate policy. I've updated to Chrome Version 41.0.2272.101 m.

Updating has gotten rid of the settings issue entirely. My old whitelist is back, and I can save settings after adjusting the automatic suspension time.

However, I still cannot suspend at all. There is no longer an error message in Inspect Popup, but I have yet to succeed at suspending tabs at all.

**Comment by Schmoba** [comment]
Hi,

I don't know whether it's related to your problem, but I can't suspend tabs either.

I started with a similar version of Chrome, but updated to the most recent. Great Suspender version is 6.14.

This is what I got from the inspect views: background page:

Uncaught (in promise) DOMException: Failed to execute 'transaction' on 'IDBDatabase': One of the specified object stores was not found. {message: "Failed to execute 'transaction' on 'IDBDatabase': One of the specified object stores was not found.", name: "NotFoundError", code: 8, stack: "Error: Failed to execute 'transaction' on 'IDBData…bkeccnjlkjkiokjodocebajanakg/js/gsUtils.js:208:19", INDEX_SIZE_ERR: 1…}code: 8message: "Failed to execute 'transaction' on 'IDBDatabase': One of the specified object stores was not found."name: "NotFoundError"stack: "Error: Failed to execute 'transaction' on 'IDBDatabase': One of the specified object stores was not found.↵    at Error (native)↵    at add (chrome-extension://klbibkeccnjlkjkiokjodocebajanakg/js/db.js:90:34)↵    at chrome-extension://klbibkeccnjlkjkiokjodocebajanakg/js/gsUtils.js:208:19"__proto__: DOMExceptiondb.js:90 addgsUtils.js:208 (anonymous function)

**Comment by deanoemcke** [comment]
@norachuga @Schmoba 
Spent  a bit of time trying to find an old version of chrome to test out your initial issues. but it turns out its quite a hard thing to find. might park that for now although it would be nice to know that it's compatible with some of the older versions.
As for your new issue, it seems you both have the same problem which is that your new indexedDb database hasn't initiated correctly. 
if you don't mind losing your saved sessions, then you can run the following commands from inside the console of the background tab. 
    indexedDB.deleteDatabase('tgs');
    gsUtils.setVersion('4.74');
then reload the extension. after this you shouldn't have any problems.

**Comment by deanoemcke** [comment]
ahh.. i think i might know what has happened. if you upgraded the extension while on an old version of chrome that did not support indexedDB then the database doesn't get created.
after upgrading chrome, your browser now supports indexedDB but you don't go through the upgrade process a second time.
the code I posted should fix this for you both. it essentially tells the extension to rerun the upgrade process.
would love to hear if this works for you.
meanwhile i'll try to think of a fix for this to prevent it from happening to anyone else.

**Comment by Schmoba** [comment]
@deanoemcke 
Great! That fixed it, thanks!

Regarding getting older versions of Chrome: try portable versions! If you don't find a sufficiently old one, I could also send you one (which is the one I first used for your extension).

**Comment by norachuga** [comment]
The two commands you posted were not succeeding for me. I apologize but I neglected to copy down the error messages I was getting! From my recollection of them, the first one was returning an object but the error and result members were getting a DOMException that said something like, 'a transaction is still pending'. The second one was getting a cannot call undefined. I looked into gsUtils.js and setVersion() is not actually a method.

**Comment by deanoemcke** [comment]
@norachuga sorry about that. i renamed that function for clarity in v6.15 to setLastVersion() which is why it didn't work for you. 
@Schmoba thanks for the great idea. i will def try out portable chrome. do you happen to know the version you were using initially?

**Comment by deanoemcke** [comment]
@Schmoba can't seem to find an old portable version of chrome. any chance you could share your copy?

**Comment by Schmoba** [comment]
@deanoemcke 
Sure, how can I best share it? Sorry if there is an obvious option on github, I just registered today ;)

**Comment by Schmoba** [comment]
Ah and it should have been version 32.0.1700.107

**Comment by deanoemcke** [comment]
Can you use drop box or google drive? Or even something like mega.

**Comment by Schmoba** [comment]
I don't have the original download from PortableApps, but compressed my exctracted folder.
Note that you need to execute GoogleChromePortable.exe and not App/chrome-bin/chrome.exe, which would open your installed version.

https://drive.google.com/file/d/0B00Xa5d3xaB9XzlvamJsMUZKZ2s/view?usp=sharing
Please let me know if you downloaded the file, then I can unshare it and delete the link.

**Comment by deanoemcke** [comment]
Thanks heaps. I've got the file. You can remove it now.

**Comment by deanoemcke** [comment]
@Schmoba I went through a debugging spree using the chrome portable version you gave me. I found a few bugs and have fixed them all. The result is a new version (v6.16) which should be fully compatible with chrome v35.
To install this version you'll need to do an uninstall / reinstall. Just make sure you unsuspend all tabs first - and note that you'll lose any saved sessions you've made.
Keen to hear if it all works well for you.
